src/factories/tokenWrapperFactory.ts

```diff
@@ -58,10 +58,10 @@ export default class TokenWrapperFactory {
    *
    * @memberof SecurityTokenWrapperFactory
    */
-  public getSecurityTokenInstanceFromAddress = async (address: string): Promise<SecurityTokenWrapper> => {
+  public getSecurityTokenInstanceFromAddress = async (address: string): Promise<SecurityTokenCommon> => {
```

Comment:
This function (and the next one) should return versioned contracts, not the common wrapper

For this, you should:
- Go to the `ContractFactory` class and implement a `getSecurityTokenVersion(address: string)` method (base yourself on the `getModuleVersion` and `getModuleFactoryVersion` methods in the same class)
- In that same class, modify the `getSecurityTokenContract` function to accept a version parameter (base yourself on `getCappedSTOContract` in that same class)
- Fetch the token version, and depending on it, instance the correct ST versioned class (you can check how it is done in the `getModuleInstance` method in the `ModuleWrapperFactoryClass` for Capped STO)